### PR TITLE
Check staleness of chainlink feed 

### LINKED ISSUE #107: Evaluate stale price handling
We need to get a cohesive picture ironed out into a doc for this topic. Consolidating info here

Oracles can be used when stale, up to a certain point.
Funding can check freshness (isValid) on MedianOracle interface

image

### PR DESCRIPTION
**Pull request by shuklaayush**
Addresses #107

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -19,7 +23,8 @@ jobs:
       - name: Install Dependencies
         run: npm install
 
-      - name: Run linter/formatter check
-        run: npm run solhint
-
+      - name: Run prettier check
+        run: npm run prettier:solidity:check
```
Comment (sajanrajdev) [inline .github/workflows/lint.yml]: You may want to revert these changes as discussed in the other PR.

File: package.json [hunk 2]
```diff
@@ -22,8 +22,9 @@
     "moment": "^2.29.3"
   },
   "scripts": {
-    "solhint": "./node_modules/.bin/solhint -f table src/**/*.sol",
-    "prettier:solidity": "./node_modules/.bin/prettier --write src/**/*.sol",
+    "solhint": "solhint -f table src/**/*.sol",
+    "prettier:solidity": "prettier --write src/**/*.sol",
+    "prettier:solidity:check": "prettier --check src/**/*.sol",
```
Comment (sajanrajdev) [inline package.json]: Same here

**Review by sajanrajdev — APPROVED** [review]
Generalizes the CtdlCvx provider and added further checks for staleness of the chainlink feed. 

LGTM

**Review by dapp-whisperer — APPROVED** [review]

**Comment by dapp-whisperer** [comment]
approved

noting that we will investigate the implicit assumption that it's guaranteed we get an answer in every "round" if something isn't broken.
`require(answeredInRound >= roundId, "Stale price");`
